# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/ffmpeg/FFmpegLibWrapper.h
# Commit: 9b17ac4800c6
# Full Hash: 9b17ac4800c6e3d2b487f965c68b4af824ed9fd4
# Author: Paul Adenot <paul@paul.cx>
# Date: 2024-02-28 04:30:28
# Regressor Bug: 1879873
# File Overlap Count: 1
# Description:
#   Bug 1879873 - Directly use tx instead of the older avfft interface. r=karlt,media-playback-reviewers
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D202156
# ==============================================================================

diff -r 2e5121a988a8 -r 9b17ac4800c6 dom/media/platforms/ffmpeg/FFmpegLibWrapper.h
--- a/dom/media/platforms/ffmpeg/FFmpegLibWrapper.h	Tue Feb 27 12:40:23 2024 +0000
+++ b/dom/media/platforms/ffmpeg/FFmpegLibWrapper.h	Tue Feb 27 12:40:23 2024 +0000
@@ -5,9 +5,9 @@
 #ifndef __FFmpegLibWrapper_h__
 #define __FFmpegLibWrapper_h__
 
-#include "FFmpegRDFTTypes.h"  // for AvRdftInitFn, etc.
 #include "mozilla/Attributes.h"
 #include "mozilla/Types.h"
+#include "ffvpx/tx.h"
 
 struct AVCodec;
 struct AVCodecContext;
@@ -148,11 +148,6 @@
   int (*avcodec_send_frame)(AVCodecContext* avctx, const AVFrame* frame);
   int (*avcodec_receive_frame)(AVCodecContext* avctx, AVFrame* frame);
 
-  // libavcodec optional
-  AvRdftInitFn av_rdft_init;
-  AvRdftCalcFn av_rdft_calc;
-  AvRdftEndFn av_rdft_end;
-
   // libavutil
   void (*av_log_set_level)(int level);
   void* (*av_malloc)(size_t size);
@@ -216,6 +211,14 @@
   void* (*vaGetDisplayDRM)(int fd);
 #endif
 
+  // Only ever used with ffvpx
+  // Unfortunately GCC rejects the following:
+  // decltype(av_tx_init)* av_tx_init;
+  // decltype(av_tx_uninit)* av_tx_uninit;
+  int (*av_tx_init)(AVTXContext **ctx, av_tx_fn *tx, enum AVTXType type,
+                    int inv, int len, const void *scale, uint64_t flags);
+  void (*av_tx_uninit)(AVTXContext **ctx);
+
   PRLibrary* mAVCodecLib;
   PRLibrary* mAVUtilLib;
 #ifdef MOZ_WIDGET_GTK